### PR TITLE
Added npm and node-js packages

### PR DESCRIPTION
**Pull request by jppelteret**
Added npm and node-js packages for use in the Jupyter suite (#1297)

### CODE REVIEW

File: var/spack/repos/builtin/packages/node-js/package.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        #     options.extend(['--with-intl=system-icu'])
+
+        configure(*options)
+
+        make('install')
+        if '+doc' in spec:
+            make('doc')
+        if self.run_tests:
+            make('test')
+            make('test-addons')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/node-js/package.py]: You'll probably want to build docs and run tests before installation.

File: var/spack/repos/builtin/packages/npm/package.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    version('3.10.5', '46002413f4a71de9b0da5b506bf1d992')
+
+    depends_on('node-js', type='build')
+
+    def install(self, spec, prefix):
+        configure('--prefix={0}'.format(prefix))
+
+        make('install')
+        if self.run_tests:
+            make('test')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/npm/package.py]: Same here, install after running tests.

**Comment by jppelteret** [comment]
@tgamblin @citibeth @adamjstewart Same with this PR please. I was able to install and use both of these packages. I have no idea why its not passing tests though. If there are **minor** changes to be made then I'm willing to do so.

**Comment by tgamblin** [comment]
It's failing on the Python 2.6 support test:

```
======================================================================
FAIL: test_package_module_compatibility (spack.test.python_version.PythonVersionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/LLNL/spack/lib/spack/spack/test/python_version.py", line 101, in test_package_module_compatibility
    self.check_python_versions(*self.pyfiles(spack.packages_path))
  File "/home/travis/build/LLNL/spack/lib/spack/spack/test/python_version.py", line 95, in check_python_versions
    self.assertTrue(len(all_issues) == 0)
AssertionError: False is not true
-------------------- >> begin captured stdout << ---------------------
File                                                           Reason
------------------------------------------------------------   --------------------
spack/var/spack/repos/builtin/packages/node-js/package.py:65   subprocess.check_output
spack/var/spack/repos/builtin/packages/node-js/package.py:66   subprocess.check_output
```

`subprocess.check_output` is a 2.7 feature and we still support 2.6.  You can probably replace the offending lines with this:

```python
if sys.platform == 'darwin':
    libtool = which('libtool')
    assert libtool.command == '/usr/bin/libtool', # ...
```

**Pull request closed without merging**